demisto_sdk/commands/validate/tests/GR_validators_test.py

```diff
@@ -529,3 +535,185 @@ def test_IsUsingUnknownContentValidator__different_dependency_type__list_files(
+    pack_2 = graph_repo.create_pack("pack2")
+    pack_2.create_playbook(
+        "UsingDeprecatedCommand", yml=playbook_dict_using_deprecate_commands
+    )
+    pack_2.create_playbook(
+        name="DeprecatedPlaybook",
+        yml={
+            "deprecated": "true",
+            "id": "DeprecatedPlaybook",
+            "name": "DeprecatedPlaybook",
+        },
+    )
+    pack_2.create_playbook(
+        name="UsingDeprecatedPlaybook", yml=playbook_dict_using_deprecated_playbook
+    )
+    pack_2.create_script(name="DeprecatedScript").set_data(**{"deprecated": "true"})
+    pack_2.create_script(
+        name="UsingDeprecatedScript",
+        code='demisto.execute_command("DeprecatedScript", dArgs)',
+    )
+    pack_2.create_script(name="SampleScript")
+    return graph_repo
+
+
+@pytest.mark.parametrize(
+    "playbook_index, expected_validation_count",
+    [
+        pytest.param(0, 1, id="Playbook using deprecated command"),
+        pytest.param(2, 1, id="Playbook using deprecated playbook"),
+    ],
+)
+def test_GR107_IsDeprecatedContentItemInUsageValidatorListFiles_invalid_playbook(
+    repo_for_test_gr_107: Repo, playbook_index: int, expected_validation_count: int
+):
+    """
+    Test the GR107_IsDeprecatedContentItemInUsageValidatorListFiles validator for invalid cases.
+
+    Given:
+    - A repository with deprecated content items in use.
+
+    When:
+    - Running the GR107_IsDeprecatedContentItemInUsageValidatorListFiles on specific playbooks.
+
+    Then:
+    - Verify that the validator correctly identifies the usage of deprecated content items.
+
+    Parameters:
+    - playbook_index: Index of the playbook to test in the pack.
+    - expected_validation_count: Expected number of validation results.
+    """
+    graph_interface = repo_for_test_gr_107.create_graph()
+    BaseValidator.graph_interface = graph_interface
+
+    pack_objects = [
+        repo_for_test_gr_107.packs[1]
+        .playbooks[playbook_index]
+        .get_graph_object(graph_interface),
+    ]
+    validator = GR107_IsDeprecatedContentItemInUsageValidatorListFiles()
+    validation_results = validator.obtain_invalid_content_items(pack_objects)
+
+    assert len(validation_results) == expected_validation_count
+
+
+def test_GR107_IsDeprecatedContentItemInUsageValidatorListFiles_invalid_script(
+    repo_for_test_gr_107: Repo,
+):
+    """
+    Test the GR107_IsDeprecatedContentItemInUsageValidatorListFiles validator for an invalid script.
+
+    Given:
+    - A repository with a script that uses a deprecated content item.
+
+    When:
+    - Running the GR107_IsDeprecatedContentItemInUsageValidatorListFiles on the specific script.
+
+    Then:
+    - Verify that the validator correctly identifies the usage of the deprecated content item.
+    - Assert that the validation results contain exactly one item.
+    """
+    graph_interface = repo_for_test_gr_107.create_graph()
+    BaseValidator.graph_interface = graph_interface
```

Comment:
is this way of mocking safe across (i.e. doesn't affect other) tests? Usually `pytest`'s `mocker` is preferred.